From 677f5c0573ff298bf40a443a721f76e45aa9bdb3 Mon Sep 17 00:00:00 2001
From: Cristina Ciocan <cristina-mihaela.ciocan@nxp.com>
Date: Mon, 30 Jan 2017 11:31:21 +0200
Subject: [PATCH 1506/5242] MLK-13840 fbdev: mxsfb: Fix explicit null
 dereference

commit  8568003734d23647acaec4056ba1374fbc99cdfa from
https://source.codeaurora.org/external/imx/linux-imx.git

In mxsfb driver, function overlayfb_check_var, a null pointer dereference
occurs if fourcc pixel format is one not considered explicitly.

This case should not occur, since the pixel format is verified against
supported values before getting to this null dereference code, but this may
change if overlay_fmt_support() or overlayfb_check_var() changes and they
are not kept in sync.

Signed-off-by: Cristina Ciocan <cristina-mihaela.ciocan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxsfb.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index 4022cae..989a952 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -1792,6 +1792,14 @@ static int overlayfb_check_var(struct fb_var_screeninfo *var,
 		case V4L2_PIX_FMT_RGB565:
 			rgb = def_rgb565;
 			break;
+		default:
+			/*
+			 * This should never be reached since the verification
+			 * is done in overlay_fmt_support(), but handle this in
+			 * case there will be a sync error between formats
+			 * supported in fmt_support and this function.
+			 */
+			 return -EINVAL;
 		}
 		break;
 	}
-- 
1.7.9.5

